pkgs/development/php-packages/grpc/default.nix

```diff
@@ -1,19 +1,28 @@
 { buildPecl
-, zlib
+, pkg-config
 , lib
+, grpc
 }:
 
 buildPecl {
   pname = "grpc";
-  version = "1.56.0";
+  version = grpc.version;
+  src = grpc.src;
 
-  sha256 = "sha256-uzxYMUzExMBDtwv3FipOuuUHg0v1wqAUtn69jXAQnf4=";
+  sourceRoot = "source/src/php/ext/grpc";
 
+  patches = [
+    ./use-pkgconfig.patch # https://github.com/grpc/grpc/pull/35404
+    ./skip-darwin-test.patch # https://github.com/grpc/grpc/pull/35403
+  ];
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ grpc ];
+
+  checkPhase = "NO_INTERACTON=yes make test";
```

Comment:
Does `checkFlags` not work for `NO_INTERACTION`? If not, we should run `pre`/`postCheck` hooks here. And drop `checkTarget` since that is redundant.

Reply:
Actually, looks like `buildPecl` already sets `checkPhase` like this, so all of these can be dropped.

`buildPecl` should probably be adjusted as suggested above. But that is for a separate PR.